data/src/main/java/com/linkedin/data/schema/PdlBuilder.java

```diff
@@ -258,6 +261,8 @@ private static String escapeIdentifier(String identifier)
    */
   abstract PdlBuilder writeJson(Object value) throws IOException;
 
+  abstract PdlBuilder writeDefaultValue(Object value, DataSchema schema) throws IOException;
```

Comment:
I recommend not introducing a method unique to default values, but rather just extend the `writeJson` method to accept a second parameter (of type `DataSchema`). For this parameter, a non-null value indicates a schema-based ordering to use, and `null` indicates to just use alphabetical ordering. This would be cleaner IMO.

Reply:
If you make this change, document this clearly in the javadoc

Reply:
That makes sense to me.